src/config/Config.zig

```diff
@@ -4493,6 +4525,32 @@ fn equalField(comptime T: type, old: T, new: T) bool {
     }
 }
 
+/// This runs a heuristic to determine if we are likely running
+/// Ghostty in a CLI environment. We need this to change some behaviors.
+/// We should keep the set of behaviors that depend on this as small
+/// as possible because magic sucks, but each place is well documented.
+fn probableCliEnvironment() bool {
```

Comment:
I put this here instead of `src/os` because its _only_ used by config defaults and I don't want this bleeding into other places.